bin/sp-data-pull

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# option to use a file. if you want to copy/paste, skip to line 9
+require 'json'
+data_file = 'your-json-file-here.json'
+imported_data = File.read(data_file)
+
+parsed_data = JSON.parse(imported_data)
+
+user_uuids = parsed_data.map { |user| user[:"properties.user_id"] }
+
+# collects user information per scenario and helps collect median information
+no_sp_users = Array.new
+single_sp_users = Array.new
+multiple_sp_users = Array.new
+
+# collect information on service provider per scenario
+single_sp_list = Set.new
+multiple_sp_list = Set.new
+
+# count of service providers
+single_sp_tally  = 0
+multiple_sp_tally = 0
+
+User.where(uuid: user_uuids).in_batches(of: 1_000) do |batch|
+  no_sp_users = batch.select { |user| user.service_providers.blank? }
+  single_sp_users = batch.select { |user| user.service_providers.count == 1 }
+  multiple_sp_users = batch.select { |user| user.service_providers.count > 1 }
```

Comment:
We might be able to avoid some excess querying here by getting the count once and then putting the user into the corresponding bucket.

```suggestion
  no_sp_users = []
  single_sp_users = []
  multiple_sp_users = []
  batch.each do |user|
    sp_count = user.service_providers.count
    no_sp_users << user if sp_count.zero?
    single_sp_users << user if sp_count == 1
    multiple_sp_users << user if sp_count > 1
  end
```